src/parquet/file/writer-internal.cc

```diff
@@ -142,7 +142,7 @@ ColumnWriter* RowGroupSerializer::NextColumn() {
   // Throws an error if more columns are being written
   auto col_meta = metadata_->NextColumnChunk();
 
-  if (current_column_writer_) { total_bytes_written_ = current_column_writer_->Close(); }
+  if (current_column_writer_) { total_bytes_written_ += current_column_writer_->Close(); }
```

Comment:
Did this below check fail for multiple columns ? 
https://github.com/apache/parquet-cpp/commit/55604b297f444e95e132658b2ef384870ae1f701#diff-553ab87bc327b48a8dcd0b499b521737R470

IIRC we are not using this value to set the rowgroup size.